collector/diskstats_common.go

```diff
@@ -72,3 +81,28 @@ var (
 		nil,
 	)
 )
+
+func newDiskstatsDeviceFilter(logger log.Logger) (deviceFilter, error) {
+	if *oldDiskstatsDeviceExclude != "" {
+		if *diskstatsDeviceExclude == "" {
+			level.Warn(logger).Log("msg", "--collector.diskstats.ignored-devices is DEPRECATED and will be removed in 2.0.0, use --collector.diskstats.device-exclude")
+			*diskstatsDeviceExclude = *oldDiskstatsDeviceExclude
+		} else {
+			return deviceFilter{}, errors.New("--collector.diskstats.ignored-devices and --collector.diskstats.device-exclude are mutually exclusive")
+		}
+	}
+
+	if *diskstatsDeviceExclude != "" && *diskstatsDeviceInclude != "" {
```

Comment:
I'm more a `len` fan guys when we want to know if a string is empty. I found it clearer and less subject to have a nasty space between the quotes. But that's maybe just my taste.

```suggestion
if len(*diskstatsDeviceExclude) > 0 && len(*diskstatsDeviceInclude) > 0 {
```

Reply:
This is the pattern we use in all other instances of this feature in other collectors. I'd rather keep to the existing code style.